core/src/test/java/org/infinispan/stream/BaseSetupStreamIteratorTest.java

```diff
@@ -90,4 +100,52 @@ public String convert(Object key, String value, Metadata metadata) {
          }
       }
    }
+
+   private static class TestDefaultConsistentHashFactory
+         extends BaseControlledConsistentHashFactory<DefaultConsistentHash> {
+      TestDefaultConsistentHashFactory() {
+         super(new DefaultTrait(), 3);
+      }
+
+      @Override
+      protected List<Address> createOwnersCollection(List<Address> members, int numberOfOwners, int segmentIndex) {
+         if (members.size() < 3)
+            return members;
+
+         switch (segmentIndex) {
+            case 0:
+               return Arrays.asList(members.get(0), members.get(1)).subList(0, numberOfOwners);
+            case 1:
+               return Arrays.asList(members.get(1), members.get(2)).subList(0, numberOfOwners);
+            case 2:
+               return Arrays.asList(members.get(2), members.get(1)).subList(0, numberOfOwners);
```

Comment:
The tests need keys with owners 2&1 and with owners 1&2. I could add another segment, but none of the `MagicKeys` generated by the tests would use it.